### PR TITLE
DNS query overview graph

### PR DESCRIPTION
**Pull request by oazizi000**
Clusters queries originating from pods and displays summary stats about successful resolutions.

### CODE REVIEW

**Comment by oazizi000** [comment]
This relies on new features in Vizier. Do not merge until next release.

**Comment by oazizi000** [comment]
Some screenshots for reference.

Screenshot from 2020-11-13 11-39-54

Screenshot from 2020-11-13 10-08-21

File: pxl_scripts/px/dns_query_summary/dns_query_summary.pxl [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    # Before aggregating, output individual requests to drawer.
+    px.debug(df, "events")
+
+    # Aggregate by each pod making a query to a particular DNS.
+    df.rcode0 = df.rcode == 0
+    df.rcode3 = df.rcode == 3
+    df = df.groupby(['pod', 'dns_server', 'qname']).agg(
+        total=('resolved', px.count),
+        resolved=('resolved', px.sum),
+        rcode0=('rcode0', px.sum),
```
Comment (htroisi) [inline pxl_scripts/px/dns_query_summary/dns_query_summary.pxl:50]: is `rcode0` used anywhere?

File: pxl_scripts/px/dns_query_summary/dns_query_summary.pxl [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    df.dns_server = px.Service(px.nslookup(df.remote_addr))
+
+    # Before aggregating, output individual requests to drawer.
+    px.debug(df, "events")
+
+    # Aggregate by each pod making a query to a particular DNS.
+    df.rcode0 = df.rcode == 0
+    df.rcode3 = df.rcode == 3
+    df = df.groupby(['pod', 'dns_server', 'qname']).agg(
+        total=('resolved', px.count),
```
Comment (htroisi) [inline pxl_scripts/px/dns_query_summary/dns_query_summary.pxl:48]: would `requests` be more descriptive than `total`?

File: pxl_scripts/px/dns_query_summary/dns_query_summary.pxl [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    df.idxz = px.find(df.req_body, ".")
+    df.qname = px.substring(df.req_body, df.idxa+8, df.idxz-df.idxa-8)
+    df = df.drop(['idxa', 'idxz'])
+
+    # Convert DNS IP to string.
+    df.dns_server = px.Service(px.nslookup(df.remote_addr))
+
+    # Apply user-filters on the data.
+    df = df[px.contains(df.pod, pod_filter)]
+    df = df[px.contains(df.dns_server, query_filter)]
```
Comment (htroisi) [inline pxl_scripts/px/dns_query_summary/dns_query_summary.pxl:43]: `df.dns_server` ---> `df.resp_body`

File: pxl_scripts/px/dns_query_summary/dns_query_summary.pxl [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    df = df[px.contains(df.qname, dns_server_filter)]
```
Comment (htroisi) [inline pxl_scripts/px/dns_query_summary/dns_query_summary.pxl:44]: `df.qname` ---> `df.dns_server`

File: pxl_scripts/px/dns_query_summary/dns_query_summary.pxl [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    df = df.groupby(['pod', 'dns_server', 'qname']).agg(
+        total=('resolved', px.count),
+        resolved=('resolved', px.sum),
+        rcode0=('rcode0', px.sum),
+        rcode3=('rcode3', px.sum),
+    )
+
+    # Compute percentages
+    df.unresolved = 100 * (df.total - df.resolved) / df.total
+    df.nx_domain = 100 * (df.rcode3) / df.total
```
Comment (htroisi) [inline pxl_scripts/px/dns_query_summary/dns_query_summary.pxl:61]: use px.Percent() instead

File: pxl_scripts/px/dns_query_summary/dns_query_summary.pxl (hunk 1 above)
Comment (oazizi000) [reply pxl_scripts/px/dns_query_summary/dns_query_summary.pxl:55]: Nope. Removing it.

File: pxl_scripts/px/dns_query_summary/dns_query_summary.pxl (hunk 2 above)
Comment (oazizi000) [reply pxl_scripts/px/dns_query_summary/dns_query_summary.pxl:53]: Renamed a bunch of variables for clarity.

**Review by zasgar — APPROVED** [review]

**Pull request merged**